### PR TITLE
Make more of the test suite run on Mac Catalyst

### PR DESCRIPTION
**Pull request by madsmtm**
Combined with https://github.com/rust-lang/rust/pull/125225, the only failing parts of the test suite are in `tests/rustdoc-js`, `tests/rustdoc-js-std` and `tests/debuginfo`. Tested with:
```console
./x test --target=aarch64-apple-ios-macabi library/std
./x test --target=aarch64-apple-ios-macabi --skip=tests/rustdoc-js --skip=tests/rustdoc-js-std --skip=tests/debuginfo tests
```

Will probably put up a PR later to enable _running_ on (not just compiling for) Mac Catalyst in CI, though not sure where exactly I should do so? `src/ci/github-actions/jobs.yml`?

Note that I've deliberately _not_ enabled stack overflow handlers on iOS/tvOS/watchOS/visionOS (see https://github.com/rust-lang/rust/issues/25872), but rather just skipped those tests, as it uses quite a few APIs that I'd be weary about getting rejected by the App Store (note that Swift doesn't do it on those platforms either).

r? @workingjubilee

CC @thomcc

@rustbot label O-ios O-apple

### CODE REVIEW

**Review by jieyouxu — APPROVED** [review]
bootstrap, compiletest and `tests/*` changes look good to me. Feel free to r=me for those changes if the libs changes look good.

**Review by workingjubilee — CHANGES_REQUESTED** [review]
Could you rename the *run-pass* `issues-{N}.rs` tests we're diffing here to include their implied subject first, since we're touching them anyway? Well, the ones you figured out what they're for, at least, and that don't have like 5 other associated files. You don't have to do anything that touches the `ENTRY_LIMIT` numbers, just the Big List Of Badly Named Tests (those are less merge-conflict-y because they're separate lines of text).

File: library/std/src/sys/pal/unix/stack_overflow.rs [hunk 1]
```diff
@@ -491,6 +491,14 @@ mod imp {
...
 }
 
+// This is intentionally not enabled on iOS/tvOS/watchOS/visionOS, as it uses
+// several symbols that might lead to rejections from the App Store, namely
+// `sigaction`, `sigaltstack`, `sysctlbyname`, `mmap`, `munmap` and `mprotect`.
+//
+// This might be overly cautious, though it is also what Swift does (and they
+// usually have fewer qualms about forwards compatibility, since the runtime
+// is shipped with the OS):
+// <https://github.com/apple/swift/blob/swift-5.10-RELEASE/stdlib/public/runtime/CrashHandlerMacOS.cpp>
```
Comment (workingjubilee) [inline library/std/src/sys/pal/unix/stack_overflow.rs:501]: And based on the exclusion notes on the `stack-probes.rs` test, we still emit binaries with `__stack_chk_guard` and such, right? And so do they? So it's less about whether the stack is protected and more about the error message.

File: tests/ui/abi/stack-probes.rs [hunk 2]
```diff
@@ -8,6 +8,10 @@
 //@ ignore-sgx no processes
 //@ ignore-fuchsia no exception handler registered for segfault
 //@ ignore-nto Crash analysis impossible at SIGSEGV in QNX Neutrino
+//@ ignore-ios Stack probes are enabled, but the SIGSEGV handler isn't
+//@ ignore-tvos Stack probes are enabled, but the SIGSEGV handler isn't
+//@ ignore-watchos Stack probes are enabled, but the SIGSEGV handler isn't
+//@ ignore-visionos Stack probes are enabled, but the SIGSEGV handler isn't
```
Comment (workingjubilee) [inline tests/ui/abi/stack-probes.rs:14]: We should probably have a version of this test that doesn't rely on std's stack overflow handler to work.

**Comment by madsmtm** [comment]
Ended up touching `ENTRY_LIMIT`, it really made sense for the `osx-frameworks.rs` test. Tell me if I should revert that.

File: tests/ui/abi/stack-probes.rs (hunk 2 above)
Comment (madsmtm) [reply tests/ui/abi/stack-probes.rs:14]: Opened in https://github.com/rust-lang/rust/issues/125390

**Review by workingjubilee — APPROVED** [review]
Yay!

**Comment by madsmtm** [comment]
Fixed the conflict, you're probably right that I shouldn't have touched `ENTRY_LIMIT`...

@rustbot ready

**Pull request merged**